packages/rs-drive-abci/src/execution/helpers.rs

```diff
@@ -245,20 +310,30 @@ where
 
         state.full_masternode_list.extend(added_masternodes);
 
-        let updated_masternodes = updated_mns
-            .iter()
-            .map(|(pro_tx_hash, masternode)| (pro_tx_hash, masternode.clone()));
-
-        updated_masternodes.for_each(|(pro_tx_hash, state_diff)| {
+        updated_mns.iter().for_each(|(pro_tx_hash, state_diff)| {
             if let Some(masternode_list_item) = state.full_masternode_list.get_mut(pro_tx_hash) {
-                if let Some(masternode_list_item) = state.hpmn_masternode_list.get_mut(pro_tx_hash)
-                {
-                    masternode_list_item.state.apply_diff(state_diff.clone());
+                if let Some(hpmn_list_item) = state.hpmn_masternode_list.get_mut(pro_tx_hash) {
```

Comment:
I would add a comment to explain why we pay attention only to these specific fields (because validators only using this information)

Reply:
done